src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -86,8 +89,12 @@ internal FeedAndEntryMaterializerAdapter(ODataFormat odataFormat, ODataReaderWra
             if (isDeltaPayload)
             {
                 // If we have a delta payload, we need the stack data structure to keep track of the read items.
-                this.materializerStateItems = new Stack<IMaterializerState>();
-            }    
+                this.deltaMaterializerStateItems = new Stack<IMaterializerState>();

```

Comment:
Why this change?

Reply:
We added another stack to track `MaterializerEntry` items. So I renamed to have a clear distinction between them.